src/main/java/org/jetbrains/kotlinx/lincheck/runner/TestThreadExecutionGenerator.java

```diff
@@ -194,16 +196,26 @@ private static void generateRun(ClassVisitor cv, Type testType, int iThread, Lis
                 mv.invokeStatic(UTILS_TYPE, UTILS_CONSUME_CPU);
             }
             // Start of try-catch block for exceptions which this actor should handle
-            Label start, end = null, handler = null, handlerEnd = null;
+            Label handler = null;
+            Label actorEnd = mv.newLabel();
+            Label start = mv.newLabel();
+            Label end = mv.newLabel();
             if (actor.getHandlesExceptions()) {
-                start = mv.newLabel();
-                end = mv.newLabel();
                 handler = mv.newLabel();
-                handlerEnd = mv.newLabel();
                 for (Class<? extends Throwable> ec : actor.getHandledExceptions())
                     mv.visitTryCatchBlock(start, end, handler, getType(ec).getInternalName());
                 mv.visitLabel(start);
             }
+            // Catch those exceptions that were not catched yet
```

Comment:
Guess, the problem you are fixing here appears in both stress and managed strategies. Let's move this code with a test to a separate pull request.

Reply:
Not obvious how to solve this problem for stress strategies...